src/server/scripts/BrokenIsles/zone_mardum.cpp

```diff
@@ -622,6 +625,234 @@ struct npc_illidari_fighting_invasion_begins : public ScriptedAI
+    SAY_KORVAS_SLAY_MORE_DEMONS             = 0,
+    SAY_SEVIS_SAY_FIND_ALLARI               = 1,
+
+    SPELL_VISUAL_KIT_SEVIS_MOUNT            = 36264,
+
+    SPELL_CAST_MOUNT_DH_FELSABER            = 200175,
+    SPELL_ASHTONGUE_FELLSABER_KILL_CREDIT   = 200254
+};
+
+enum ETIAshtongueWaypoints
+{
+    PATH_KAYN_SUNFURY_NEAR_TELEPORT         = 9822900,
+    PATH_KORVAS_BLOODTHORN_NEAR_TELEPORT    = 9835400,
+    PATH_SEVIS_BRIGHTFLAME_GATEWAY          = 9991600
+};
+
+// 98229 - Kayn Sunfury
+struct npc_kayn_sunfury_ashtongue_intro : public ScriptedAI
+{
+    npc_kayn_sunfury_ashtongue_intro(Creature* creature) : ScriptedAI(creature) { }
+
+    void OnQuestAccept(Player* player, Quest const* quest) override
+    {
+        if (quest->GetQuestId() == QUEST_ENTER_THE_ILLIDARI_ASHTONGUE)
+        {
+            PhasingHandler::OnConditionChange(player);
+            Creature* kaynObject = GetClosestCreatureWithOptions(player, 10.0f, { .CreatureId = NPC_KAYN_SUNFURY_ASHTONGUE, .IgnorePhases = true });
+            Creature* korvasObject = GetClosestCreatureWithOptions(player, 10.0f, { .CreatureId = NPC_KORVAS_BLOODTHORN_ASHTONGUE, .IgnorePhases = true });
+            if (!kaynObject || !korvasObject)
+                return;
+
+            TempSummon* kaynClone = kaynObject->SummonPersonalClone(kaynObject->GetPosition(), TEMPSUMMON_MANUAL_DESPAWN, 0s, 0, 0, player);
+            TempSummon* korvasClone = korvasObject->SummonPersonalClone(korvasObject->GetPosition(), TEMPSUMMON_MANUAL_DESPAWN, 0s, 0, 0, player);
+            if (!kaynClone || !korvasClone)
+                return;
+
+            korvasClone->SetEmoteState(EMOTE_STATE_READY1H);
+            kaynClone->RemoveNpcFlag(NPCFlags(UNIT_NPC_FLAG_QUESTGIVER));
+        }
+    }
+};
+
+struct npc_kayn_sunfury_ashtongue_intro_private : public ScriptedAI
+{
+    npc_kayn_sunfury_ashtongue_intro_private(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        Creature* korvasObject = GetClosestCreatureWithOptions(me, 10.0f, { .CreatureId = NPC_KORVAS_BLOODTHORN_ASHTONGUE, .IgnorePhases = true, .PrivateObjectOwnerGuid = me->GetPrivateObjectOwner()});
+        if (!korvasObject)
+            return;
+
+        ObjectGuid korvasGuid = korvasObject->GetGUID();
+
+        _scheduler.Schedule(1s, [this, korvasGuid](TaskContext /*task*/)
+        {
+            Unit* privateObjectOwner = ObjectAccessor::GetUnit(*me, me->GetPrivateObjectOwner());
+            if (!privateObjectOwner)
+                return;
+
+            Unit* korvas = ObjectAccessor::GetUnit(*me, korvasGuid);
+            if (!korvas)
+                return;
+
+            Talk(SAY_KAYN_ACTIVATE_GATEWAY, me);
+            me->CastSpell(privateObjectOwner, SPELL_TRACK_TARGET_IN_CHANNEL, false);
+            korvas->CastSpell(privateObjectOwner, SPELL_TRACK_TARGET_IN_CHANNEL, false);
+        });
+
+        _scheduler.Schedule(7s, [this](TaskContext /*task*/)
```

Comment:
if there is a lag, all of this is desynchronized, you must schedule every action after the next, not altogether